sqlmesh/core/snapshot/evaluator.py

```diff
@@ -800,8 +800,8 @@ def insert(
         **kwargs: t.Any,
     ) -> None:
         if isinstance(model.kind, IncrementalUnmanagedKind) and model.kind.insert_overwrite:
-            self.adapter.insert_overwrite(
-                name, query_or_df, columns_to_types=model.columns_to_types
+            self.adapter.insert_overwrite_by_partition(
+                name, query_or_df, model.partitioned_by, columns_to_types=model.columns_to_types
```

Comment:
I'm sure this has been asked before, but just confirming that a model in dbt must have a partition_by if it is incremental?

Reply:
We validate this here: https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/model/definition.py#L626-L634